src/tools/auth0/handlers/logStreams.ts

```diff
@@ -54,29 +54,39 @@ export default class LogStreamsHandler extends DefaultAPIHandler {
       return this.existing;
     }
 
-    const logStreams = await this.client.logStreams.getAll({ paginate: false });
+    const logStreams = await this.client.logStreams
+      .getAll({ paginate: false })
+      .then((logStreams) => {
+        console.log(logStreams);
+        return logStreams.map((logStream) => {
+          if (logStream.status === 'suspended') delete logStream.status;
+          return logStream;
+        });
+      });
 
-    const nonSuspendedLogStreams = logStreams.filter(
-      (logStream: LogStream) => logStream.status !== 'suspended'
-    );
+    this.existing = logStreams;
 
-    this.existing = nonSuspendedLogStreams;
-
-    return nonSuspendedLogStreams;
+    return logStreams;
   }
 
   async processChanges(assets: Assets): Promise<void> {
     const { logStreams } = assets;
-    // Do nothing if not set
+
     if (!logStreams) return;
-    // Figure out what needs to be updated vs created
+
     const changes = await this.calcChanges(assets).then((changes) => {
       return {
         ...changes,
         update: changes.update.map((update: LogStream) => {
           if (update.type === 'eventbridge' || update.type === 'eventgrid') {
             delete update.sink;
           }
+          if (update.status === 'suspended') {
+            // @ts-ignore because while status is usually expected for update payloads, it is ok to be omitted
+            // for suspended log streams. Setting as `active` in these instances would probably be ok
+            // but bit presumptuous, let suspended log streams remain suspended.
+            delete update.status;
```

Comment:
This is an important line of code. It strips the `status: "suspended"` from being written to local configuration files when dumping a tenant. This is because you cannot apply that state on update or create, also we do not know the intended "real" state of the log stream, wether it is paused or active.

Reply:
Would it perhaps improve the DX if we do end up writing the status of the dumped logstream as suspended but instead when pushing to the Auth0 Tenant we skip the status field if it is equal to suspended and also give a warning to the user that they were trying to update/create a suspended logstream?

Reply:
That would technically work, and the functional behavior would be near-identical to what's proposed. 

The `suspended` status isn't a configuration state like `active` or `paused` is, so would recording it give the illusion that it is a configurable state? My feeling is yes because these are resource configuration files after all. Also, in a multi-tenant context, does it seem weird to promote a suspended log stream from staging=>prod? 

But then again, it may be weird to not record the state at all. But at the very least, we don't attempt to guess about the real status. And in either case, once a suspended log stream is unsuspended, its "real" status will overwrite. 

So I guess this is a tossup. Especially since there's not a big functional difference. Considering the above, which do you think is a better DX?

Additionally, to your point about improved DX, I think the one thing we could do is to set a suspended log stream as active when updated, but that seems a bit presumptuous. Or another thing we can do is

Reply:
You're right, functionally there isn't much difference. The main point I was considering is letting the user know that they had a suspended log stream and that they're:

- [same tenant] - updating all the other fields except status
- [other tenant] - creating a new log stream that is paused 

Otherwise this will already happen behind the scenes without any information. Although maybe the extra warning would just risk causing confusion. 🤔 So I'm fine either way, it was mostly something to think about for me as well.